### PR TITLE
Heroku

### PR DESCRIPTION
**Pull request by fgregg**
Following instructions from: https://github.com/datamade/erikson-edi/pull/162/

### CODE REVIEW

**Comment by derekeder** [comment]
looks like we can get the heroku connection string this way:

`heroku pg:credentials DATABASE`

documentation: https://devcenter.heroku.com/articles/connecting-heroku-postgres

**Comment by derekeder** [comment]
Steps for restoring the production database on heroku:

* set up and configure the Heroku app. provision the `standard-0` database (we need the space)
* deploy the `heroku` branch
* reset the database: `heroku pg:reset --app chi-councilmatic-staging`
* use `psql` to execute the database dump: `cat ~/Downloads/chicago_prod_2022_07_01.dump | heroku pg:psql --app chi-councilmatic-staging`

**Comment by derekeder** [comment]
staging site deployed with a copy of the production database from July 1, 2022: https://chi-councilmatic-staging.herokuapp.com/

Next step is to set up the Elastisearch addon. Looks like Heroku offers several options, all of which cost money at certain sizes: https://elements.heroku.com/search/addons?q=elastisearch

Searchbox looks to be the cheapest, but it depends on how much space we need. @hancush I see the elastisearch container has a 512m heap size. If that's sufficient, we can probably try out the $9/mo micro plan

**Comment by hancush** [comment]
@derekeder searchbox must be new, because it didn't come up during my evaluation of add-ons. i recommended bonsai because it's most mature and reasonably priced. https://docs.google.com/document/d/1VTDFenEbG9gMJkf6kpBD7CMMgeU6_Ia71c0moziYnbM/edit?usp=sharing

**Comment by derekeder** [comment]
I compared Searchbox to Bonsai and the big difference is in number of documents. We have 130,000 bills (and growing) which puts us at the $50/mo plan for Bonsai. Looks like we can use the $10/mo plan for Searchbox (no document limit, just space), so I'm going to try that.

**Comment by derekeder** [comment]
Currently, Elasticsearch is only showing us the top 10,000 search results. This happens both locally and on Heroku. To fix this, we tried a few things, but no dice.

The default setting for elasticsearch [index.max_result_window](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html#index-max-result-window) is 10,000.

We tried:
- updating the `ENV` variables in the docker container to set `index.max_result_window` to a higher value. this caused the container to crash. We also updated `"ES_JAVA_OPTS=-Xms2048m -Xmx2048m"`, which also caused the container to crash.
- getting a page count via other means. the current template uses `paginator.count`. We tried `form.searchqueryset.count` but it yields the same result

This may be a setting we could change with [Bonsai](https://devcenter.heroku.com/articles/bonsai#installing-the-add-on), but we didn't find a definitive answer to that in the docs.

**Comment by derekeder** [comment]
Update after our hacking session today:

* looking at this again, the 10,000 entry limit seems to not really be an issue anymore. the facet counts seem to be working as expected now. the council members list does seem to be incomplete, but this may be due to a different issue. so, moving on ...
* successfully connected to the heroku database from the ocd server using the heroku connection string `heroku pg:credentials:url DATABASE`
* we started to set up a repository to run our Chicago scrapers using GitHub actions: https://github.com/datamade/chicago-council-scrapers. With this, we will be able to retire ocd.datamade.us, and because its open source, will run for free on GH Actions.

**Comment by derekeder** [comment]
latest update - our final blocker with bringing this is has been resolved.

the number of facets returned for sponsors was limited to 10, which is the default setting in elasticsearch: https://www.elastic.co/guide/en/app-search/7.14/facets.html

we were able to override this setting in the `get_queryset` function in `ChicagoCouncilmaticFacetedSearchView` like so:

```python
def get_queryset(self):
        sqs = super().get_queryset()

        # default facet list size is 10, but we want to show all
        options = {
            "size": 250
        }
        for field in self.facet_fields:
            sqs = sqs.facet(field, **options)
```

**Comment by derekeder** [comment]
@fgregg I'm good with merging this in, you?

File: chicago/views.py [hunk 1]
```diff
@@ -171,63 +173,88 @@ def get_seo_blob(self):
...
+        'bill_type',
+        'sponsorships',
+        'controlling_body',
+        'inferred_status',
+        'topics',
+        'legislative_session',
+    ]
 
-        form_kwargs['selected_facets'] = self.request.GET.getlist("selected_facets")
+    def dispatch(self, *args, **kwargs):
```
Comment (fgregg) [inline chicago/views.py:189]: this method is not doing anything. let's remove it.

File: scripts/docker-entrypoint.sh [hunk 2]
```diff
@@ -3,8 +3,8 @@ set -e
 
 if [ "$DJANGO_MANAGEPY_MIGRATE" = "True" ]; then
     python manage.py migrate --noinput
-    python manage.py import_shapes data/final/shapes/chicago_shapes.json
-    python manage.py collectstatic --noinput
+    # python manage.py import_shapes data/final/shapes/chicago_shapes.json
+    # python manage.py collectstatic --noinput
```
Comment (fgregg) [inline scripts/docker-entrypoint.sh:7]: i think this needs to be moved to release.sh?

File: scripts/docker-entrypoint.sh [hunk 3]
```diff
@@ -3,8 +3,8 @@ set -e
 
 if [ "$DJANGO_MANAGEPY_MIGRATE" = "True" ]; then
     python manage.py migrate --noinput
-    python manage.py import_shapes data/final/shapes/chicago_shapes.json
-    python manage.py collectstatic --noinput
+    # python manage.py import_shapes data/final/shapes/chicago_shapes.json
```
Comment (fgregg) [inline scripts/docker-entrypoint.sh:6]: uncomment or delete.

**Review by fgregg — COMMENTED** [review]
A few changes we need to make before bringing this in.

**Pull request merged**